### PR TITLE
[OpenAL] Move our listener info out of platform listener object (Fix for #5797)

### LINKED ISSUE #5797: exception thrown: TypeError: undefined is not an object (evaluating 'AL.currentContext.ctx.listener._position[0]=v1'),_alListener3f
When I run my demo
http://esenthel.com/site/Esenthel%20Fantasy%20Demo/Esenthel%20Fantasy%20Demo.html
using Safari browser, I get this error:
```
[Log] exception thrown: TypeError: undefined is not an object (evaluating 'AL.currentContext.ctx.listener._position[0]=v1'),_alListener3f@http://esenthel.com/site/Esenthel%20Fantasy%20Demo/Esenthel%20Fantasy%20Demo.js:1:208366 (Esenthel Fantasy Demo.html, line 26)
wasm function@[wasm code]
wasm function: 4569@[wasm code]
wasm function: 4557@[wasm code]
wasm function: 2092@[wasm code]
wasm function: 4353@[wasm code]
wasm function: 4351@[wasm code]
wasm function: 6291@[wasm code]
wasm function@[wasm code]
dynCall_v@[native code]
http://esenthel.com/site/Esenthel%20Fantasy%20Demo/Esenthel%20Fantasy%20Demo.js:1:373816
browserIterationFunc@http://esenthel.com/site/Esenthel%20Fantasy%20Demo/Esenthel%20Fantasy%20Demo.js:1:174917
runIter@http://esenthel.com/site/Esenthel%20Fantasy%20Demo/Esenthel%20Fantasy%20Demo.js:1:177994
Browser_mainLoop_runner@http://esenthel.com/site/Esenthel%20Fantasy%20Demo/Esenthel%20Fantasy%20Demo.js:1:176459
```

Is this a bug in Safari?
Because Chrome, Firefox and Opera work fine.
Edge has another bug..

### PR DESCRIPTION
**Pull request by hgy29**
Checked with firefox and iOS safari, still working ok with the former and no more crash on the latter.
Fix #5797

### CODE REVIEW

File: src/library_openal.js [hunk 1]
```diff
@@ -532,12 +532,12 @@ var LibraryOpenAL = {
       // a displacement from the listener.
       if (src.relative) {
         // Negate the listener direction since forward is -Z.
-        var lBackX = -listener._direction[0];
-        var lBackY = -listener._direction[1];
-        var lBackZ = -listener._direction[2];
-        var lUpX = listener._up[0];
-        var lUpY = listener._up[1];
-        var lUpZ = listener._up[2];
+        var lBackX = listener.direction[0];
```
Comment (kripken) [inline src/library_openal.js]: why did the `-` negations go away?

File: src/library_openal.js [hunk 2]
```diff
@@ -468,29 +468,29 @@ var LibraryOpenAL = {
...
+        listener.upY.value = ctx.listener.up[1];
+        listener.upZ.value = ctx.listener.up[2];
       } else {
 #if OPENAL_DEBUG
         warnOnce('Listener orientation attributes are not present, falling back to setOrientation()');
 #endif
         listener.setOrientation(
-          listener._direction[0], listener._direction[1], listener._direction[2],
-          listener._up[0], listener._up[1], listener._up[2]);
+        		ctx.listener.direction[0], ctx.listener.direction[1], ctx.listener.direction[2],
```
Comment (kripken) [inline src/library_openal.js]: indentation looks wrong here

**Comment by kripken** [comment]
Thanks! Looks good aside from those comments. Please also add yourself to AUTHORS, and verify that `./tests/runner.py interactive.test_openal*` pass (we can't run those on CI automatically, sorry about that - and let me know if you see any problems there).

File: src/library_openal.js (hunk 1 above)
Comment (hgy29) [reply src/library_openal.js]: My mistake, must have been too fast at removing the underscores from member names. Thanks for spotting this, I'll fix it.

File: src/library_openal.js (hunk 2 above)
Comment (hgy29) [reply src/library_openal.js]: Indeed, looks like my editor wasn't setup properly for 2 spaces indentations

**Comment by hgy29** [comment]
Checked with `./tests/runner.py interactive.test_openal*`: all ok, except for microphone test, since I don't have a usable one.

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**